python/ray/dag/compiled_dag_node.py

```diff
@@ -765,7 +765,18 @@ def _preprocess(self) -> None:
         self.input_task_idx, self.output_task_idx = None, None
         self.actor_task_count.clear()
 
-        nccl_actors_p2p: Set["ray.actor.ActorHandle"] = set()
+        # Actors and DAG nodes involved in NCCL P2P send/recv with `transport=nccl`,
```

Comment:
The logic in this block of code is getting quite complicated. Let's try to clean it up a bit:
- Separate it out into a helper function.
- Ideally, we do it in these steps and none of them should be interleaved:
1. Collect mapping of set(actors) <> NCCL group specified (None for "default").
2. Validate all groups. I.e. if NCCL group was specified, check that actors are actually in that group
3. Initialize groups if needed.
4. Set type hints on DAG nodes.

Reply:
Factored steps 2-4 out into a helper function. Kept step 1 in `_preprocess`.